### PR TITLE
feat(rspack_plugin_css): add css/auto same as webpack

### PR DESCRIPTION
**Pull request by LingyuCoder**
## Summary

According to the issue #2273, equivalent implementation with webpack pull request[ #16572](https://github.com/webpack/webpack/issues/16572)

### CODE REVIEW

**Comment by hyf0** [comment]
Webpak doesn't think this as a breaking change. Do we need to? @hardfist

File: crates/rspack_plugin_css/src/parser_and_generator/mod.rs [hunk 1]
```diff
@@ -33,6 +35,9 @@ use crate::swc_css_compiler::{SwcCssSourceMapGenConfig, SWC_COMPILER};
 use crate::utils::{css_modules_exports_to_string, ModulesTransformConfig};
 use crate::{pxtorem::px_to_rem::px_to_rem, visitors::analyze_dependencies};
 
+static REGEX_IS_MODULES: Lazy<Regex> =
+  Lazy::new(|| Regex::new(r"\.module(s)?\.[^.]+$").expect("Invalid regex"));
```
Comment (hyf0) [inline crates/rspack_plugin_css/src/parser_and_generator/mod.rs:39]: So `module.scss` would be considered as CSS module. Does Webpack do the same thing?

Comment (ahabhgk) [reply crates/rspack_plugin_css/src/parser_and_generator/mod.rs:39]: with css/auto, yes

Comment (LingyuCoder) [reply crates/rspack_plugin_css/src/parser_and_generator/mod.rs:39]: > So `module.scss` would be considered as CSS module. Does Webpack do the same thing?

Webpack use RegExp in module.rules.test field to specific css files. By default, webpack does not have any rule for scss files, only css default rules in webpack/lib/config/defaults.js.

File: crates/rspack_plugin_css/tests/fixtures/custom/css_auto/expected/main.css [hunk 2]
```diff
@@ -0,0 +1,6 @@
+.style {
```
Comment (ahabhgk) [inline crates/rspack_plugin_css/tests/fixtures/custom/css_auto/expected/main.css:1]: avoid to add tests on rust side, we lack of human maintenance and js side test is good enough in most of cases

**Review by ahabhgk — APPROVED** [review]

**Review by hyf0 — APPROVED** [review]

**Pull request merged**